test/framework/deployments/postgres/kubernetes.go

```diff
@@ -0,0 +1,56 @@
+package postgres
+
+import (
+	"github.com/gruntwork-io/terratest/modules/helm"
+
+	"github.com/kumahq/kuma/test/framework"
+)
+
+type k8SDeployment struct {
+	envVars map[string]string
+	options *deployOptions
+}
+
+func (t *k8SDeployment) GetEnvVars() map[string]string {
+	return t.envVars
+}
+
+var _ PostgresDeployment = &k8SDeployment{}
+
+func (t *k8SDeployment) Name() string {
+	return t.options.deploymentName
+}
+
+func (t *k8SDeployment) Deploy(cluster framework.Cluster) error {
+	releaseName := "postgres-release"
+
+	helmOpts := &helm.Options{
+		SetValues: map[string]string{
+			"global.postgresql.auth.username": t.options.username,
+			"global.postgresql.auth.password": t.options.password,
+			"global.postgresql.auth.database": t.options.database,
+			"postgresql.primary.fullname":     t.options.primaryName,
+		},
+		KubectlOptions: cluster.GetKubectlOptions(t.options.namespace),
+	}
+
+	err := helm.AddRepoE(cluster.GetTesting(), helmOpts, "bitnami", "https://charts.bitnami.com/bitnami")
+	if err != nil {
+		return err
+	}
+
+	return helm.InstallE(cluster.GetTesting(), helmOpts, "bitnami/postgresql", releaseName)
+}
+
+func (t *k8SDeployment) Delete(cluster framework.Cluster) error {
+	if !t.options.skipNamespaceCleanup {
+		return cluster.DeleteNamespace(t.options.namespace)
+	}
```

Comment:
It's weird that we do not create a namespace in deployment, but we are trying to delete it.

I'd either do noop cleanup or delete helm release

Reply:
We do it manually here https://github.com/kumahq/kuma/blob/e0a2b06a7121f4e5a6c1df00c9ee2b269d7dd9d1/test/e2e/helm/kuma_helm_deploy_global_universal_mode_and_zone.go#L48 because we need to put the secret before we deploy kuma and every "Install" is coupled with "Delete", but on  https://github.com/kumahq/kuma/blob/e0a2b06a7121f4e5a6c1df00c9ee2b269d7dd9d1/test/e2e/helm/kuma_helm_deploy_global_universal_mode_and_zone.go#L122 we delete the namespace with the whole deployment so without this "skip" the test fails in `E2EAfterAll` because there is no "kuma-namespace". If there is a better way to do this without this skip then I'm all for it.

Reply:
Yes, I know that we create the namespace manually.

What I mean is that the `k8SDeployment` should not delete namespace if it's not creating this namespace in `Deploy`. It should leave managing the namespace for the client of this object.
`Delete` could however delete HELM deployment without deleting the namespace. However, I'm ok with not implementing it if you want to skip it because we usually delete namespace anyways.

Reply:
done